File: prototype/rails/src/app/models/payout_test.rb
require "commerce_test_case"

class PayoutTest < CommerceTestCase
  def test_a_payout_reads_its_amount_in_money
    payout = Payout.create!(
      seller: seller, period_start: Date.new(2026, 8, 17), period_end: Date.new(2026, 8, 23),
      amount_cents: 40_500, paid_at: moment("2026-08-24 09:00:00")
    )

    assert_equal "$405.00", payout.amount.format
  end
end

File: prototype/php/src/app/Models/PayoutTest.php
<?php

declare(strict_types=1);

namespace App\Models;

it('reads its amount as money', function (): void {
    $payout = Payout::create([
        'seller_id' => $this->seller()->id,
        'period_start' => '2026-08-10',
        'period_end' => '2026-08-16',
        'amount_cents' => 9000,
        'paid_at' => '2026-08-17 00:00:00',
    ]);

    expect($payout->amount())->toBeMoney(9000);
});

it('narrows the list to one seller, or to everyone when the filter is empty', function (): void {
    $matching = $this->seller('Blue Kiln Studio');
    $other = $this->seller('Rye Press');
    Payout::create(['seller_id' => $matching->id, 'period_start' => '2026-08-10', 'period_end' => '2026-08-16', 'amount_cents' => 9000, 'paid_at' => '2026-08-17 00:00:00']);
    Payout::create(['seller_id' => $other->id, 'period_start' => '2026-08-10', 'period_end' => '2026-08-16', 'amount_cents' => 5000, 'paid_at' => '2026-08-17 00:00:00']);

    expect(Payout::query()->ofSeller($matching->id)->pluck('seller_id')->all())->toBe([$matching->id])
        ->and(Payout::query()->ofSeller(null)->count())->toBe(2);
});
